resources/docker/README.md

```diff
@@ -0,0 +1,169 @@
+after first build. If changes are made, it can be rebuild using `docker-compose build <container> --no-cache`.
+* To rebuild all containers on startup use the `--force-recreate` flag as such: `docker-compose up --force-recreate`
+* If a container persists still, use `docker-compose rm <container_id>` to remove it from the docker-compose cache and
+then recreate the containers.
+
+## Override default configuration
+In your local environment, on the project root, you can create a second docker-compose yml file called
+`docker-compose.local.yml`. This file is ignored by git. In that file, you can set up services overrides and setup your
+own settings on the environment. Below is an example of the docker-compose.override.yml that allows the user that runs
+apache to have its UID and GID changed (the group's id changes) and sets up the server to run on port 80.
+```yaml
+version: '2'
+services:
+  web:
+    build:
+      args:
+        DAEMON_UID: "1000"
+        DAEMON_GID: "1000"
+    expose:
+      - "80"
+    ports:
+      - "80:80"
+    environment:
+      PORT: "80"
+```
+The rest of the service will get the properties from the original composer file.
+
+To run the containers by reading all overrides, use the following command: `docker-composer -f docker-compose.yml
+-f docker-compose.local.yml up`. Please, note, that the last file in the command has bigger priority. More than one
+overrides can be provided.
+
+As an example of usage, if you want to have your personal git, composer and ssh settings persist over the web container,
+you would have to have something like the following override file:
+```yaml
+version: '3.4'
+services:
+  web:
+    volumes:
+      - .:/var/www/html
+      - ~/.gitconfig:/var/www/.gitconfig
+      - ~/.gitignore:/var/www/.gitignore
+      - ~/.composer:/var/www/.composer
+      - ~/.ssh:/var/www/.ssh
+```
+Note that in the above piece, the `- .:/var/www/html` is also part of the main `docker-composer.yml`. This is because
+the `volumes` entry here will completely override the parent entry and will not merge with it.
+
+## Rebuild from existing databases
+By default, the phing target `download-databases` sends the downloaded databases to the `tmp` folder of the project
+root. The virtuoso dumps are stored in the sub directory `dump-virtuoso` and the mysql dump is located in the `tmp`
+folder itself.
+In the project root, there is also the docker-compose.prod_db.yml which contains overrides for restoring the databases
+according to the image requirements. What it does, is that it maps the dump.sql (the mysql dump) within the startup
+directory of the mysql image, and the virtuoso dumps within the startup directory of the virtuoso image.
+
+To start the machines with the databases restored, run the following command:  
+`docker-compose -f docker-compose.yml -f docker-compose.prod_db.yml up`  
+and the images will be started.
```

Comment:
I have extended the readme a bit including a small introduction to start and stop the containers. Generally, I don't think that we need to explain every time that the containers need to be down since the `up` command is what is supposed to be starting them.

As for your case, I have included in the docker readme a small explanation that of course the database dumps need to be downloaded manually and placed according to the docker-compose.prod_db.yml.

Reply:
It's better now but this section still needs work I think. It is really confusing for newcomers. It starts with explaining a whole bunch of technical details about concepts which a new user is not familiar with.